src/core/agent/services/credentialService.ts

```diff
@@ -127,16 +135,50 @@ class CredentialService extends AgentService {
   async deleteCredential(id: string): Promise<void> {
     const metadata = await this.getMetadataById(id);
     this.validArchivedCredential(metadata);
-    // We only soft delete because we need to sync with KERIA. This will prevent re-sync deleted records.
+
     await this.credentialStorage.updateCredentialMetadata(id, {
-      isDeleted: true,
+      pendingDeletion: true,
     });
   }
 
   async deleteStaleLocalCredential(id: string): Promise<void> {
     await this.credentialStorage.deleteCredentialMetadata(id);
   }
 
+  async deleteCredentialOnKeria(id: string): Promise<void> {
+    const metadata = await this.getMetadataById(id);
+    const identifier = await this.props.signifyClient
+      .identifiers()
+      .get(metadata.identifierId);
+
+    if(metadata.pendingDeletion == true){
+      await this.props.signifyClient
+        .credentials()
+        .revoke(identifier.name, metadata.id)
```

Comment:
Revoke? It's `.delete`. I sent you the PR with the new functionality which you haven't updated to: https://github.com/WebOfTrust/signify-ts/pull/297/files

You cannot revoke a credential that you didn't issue. Have you actually tried running this code?